[Code Bug]: Spurious failures in TestCalibrateDisplayShape.{test_xy_screen_fraction,test_xyz_screen_coords}

### Contact Details

eharvey@sandia.gov

### What happened?

These two tests started failing with size mismatches on January 7th.

### Version

[develop@504d299b65f41bba527ae0c8d7e4bbdb03141a18](https://github.com/sandialabs/OpenCSP/tree/504d299b65f41bba527ae0c8d7e4bbdb03141a18)

### What OS are you seeing this on?

Windows

### Relevant dependency versions

```shell
Run python3 -m pip install -r ./OpenCSP/requirements.txt
  python3 -m pip install -r ./OpenCSP/requirements.txt
  choco install ffmpeg
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
  env:
    pythonLocation: C:\hostedtoolcache\windows\Python\3.10.11\x64
    PKG_CONFIG_PATH: C:\hostedtoolcache\windows\Python\3.10.11\x64/lib/pkgconfig
    Python_ROOT_DIR: C:\hostedtoolcache\windows\Python\3.10.11\x64
    Python2_ROOT_DIR: C:\hostedtoolcache\windows\Python\3.10.11\x64
    Python3_ROOT_DIR: C:\hostedtoolcache\windows\Python\3.10.11\x64
Collecting h5py>=3.10.0 (from -r ./OpenCSP/requirements.txt (line 12))
  Downloading h5py-3.12.1-cp310-cp310-win_amd64.whl.metadata (2.5 kB)
Collecting imageio>=2.34.0 (from -r ./OpenCSP/requirements.txt (line 13))
  Downloading imageio-2.36.1-py3-none-any.whl.metadata (5.2 kB)
Collecting matplotlib<=3.8.4 (from -r ./OpenCSP/requirements.txt (line 14))
  Downloading matplotlib-3.8.4-cp310-cp310-win_amd64.whl.metadata (5.9 kB)
Collecting pandas>=2.2.1 (from -r ./OpenCSP/requirements.txt (line 15))
  Downloading pandas-2.2.3-cp310-cp310-win_amd64.whl.metadata (19 kB)
Collecting pillow>=10.2.0 (from -r ./OpenCSP/requirements.txt (line 16))
  Downloading pillow-11.1.0-cp310-cp310-win_amd64.whl.metadata (9.3 kB)
Collecting pyproj>=3.6.1 (from -r ./OpenCSP/requirements.txt (line 17))
  Downloading pyproj-3.7.0-cp310-cp310-win_amd64.whl.metadata (31 kB)
Collecting pytz>=2024.1 (from -r ./OpenCSP/requirements.txt (line 18))
  Downloading pytz-2024.2-py2.py3-none-any.whl.metadata (22 kB)
Collecting scipy>=1.12.0 (from -r ./OpenCSP/requirements.txt (line 19))
  Downloading scipy-1.15.0-cp310-cp310-win_amd64.whl.metadata (60 kB)
Collecting sympy>=1.12 (from -r ./OpenCSP/requirements.txt (line 20))
  Downloading sympy-1.13.3-py3-none-any.whl.metadata (12 kB)
Collecting ipykernel>=6.29.3 (from -r ./OpenCSP/requirements.txt (line 23))
  Downloading ipykernel-6.29.5-py3-none-any.whl.metadata (6.3 kB)
Collecting opencv-contrib-python<4.10.0.84,>=4.8.1.78 (from -r ./OpenCSP/requirements.txt (line 24))
  Downloading opencv_contrib_python-4.10.0.82-cp37-abi3-win_amd64.whl.metadata (20 kB)
Collecting numpy<2.0.0 (from -r ./OpenCSP/requirements.txt (line 25))
  Downloading numpy-1.26.4-cp310-cp310-win_amd64.whl.metadata (61 kB)
Collecting pytest>=8.1.1 (from -r ./OpenCSP/requirements.txt (line 26))
  Downloading pytest-8.3.4-py3-none-any.whl.metadata (7.5 kB)
Collecting python-pptx>=0.6.23 (from -r ./OpenCSP/requirements.txt (line 27))
  Downloading python_pptx-1.0.2-py3-none-any.whl.metadata (2.5 kB)
Collecting rawpy>=0.19.1 (from -r ./OpenCSP/requirements.txt (line 28))
  Downloading rawpy-0.24.0-cp310-cp310-win_amd64.whl.metadata (6.4 kB)
Collecting openpyxl>=3.1.2 (from -r ./OpenCSP/requirements.txt (line 29))
  Downloading openpyxl-3.1.5-py2.py3-none-any.whl.metadata (2.5 kB)
Collecting pypylon>=2.3.0 (from -r ./OpenCSP/requirements.txt (line 32))
  Downloading pypylon-4.1.0-cp39-abi3-win_amd64.whl.metadata (9.3 kB)
Collecting pysolar>=0.11 (from -r ./OpenCSP/requirements.txt (line 34))
  Downloading pysolar-0.13-py3-none-any.whl.metadata (358 bytes)
Collecting tqdm>=4.66.2 (from -r ./OpenCSP/requirements.txt (line 35))
  Downloading tqdm-4.67.1-py3-none-any.whl.metadata (57 kB)
Collecting pytest-xvfb>=3.0.0 (from -r ./OpenCSP/requirements.txt (line 37))
  Downloading pytest_xvfb-3.0.0-py3-none-any.whl.metadata (4.8 kB)
Collecting pytest-cov>=4.1.0 (from -r ./OpenCSP/requirements.txt (line 38))
  Downloading pytest_cov-6.0.0-py3-none-any.whl.metadata (27 kB)
Collecting contourpy>=1.0.1 (from matplotlib<=3.8.4->-r ./OpenCSP/requirements.txt (line 14))
  Downloading contourpy-1.3.1-cp310-cp310-win_amd64.whl.metadata (5.4 kB)
Collecting cycler>=0.10 (from matplotlib<=3.8.4->-r ./OpenCSP/requirements.txt (line 14))
  Downloading cycler-0.12.1-py3-none-any.whl.metadata (3.8 kB)
Collecting fonttools>=4.22.0 (from matplotlib<=3.8.4->-r ./OpenCSP/requirements.txt (line 14))
  Downloading fonttools-4.55.3-cp310-cp310-win_amd64.whl.metadata (168 kB)
Collecting kiwisolver>=1.3.1 (from matplotlib<=3.8.4->-r ./OpenCSP/requirements.txt (line 14))
  Downloading kiwisolver-1.4.8-cp310-cp310-win_amd64.whl.metadata (6.3 kB)
Collecting packaging>=20.0 (from matplotlib<=3.8.4->-r ./OpenCSP/requirements.txt (line 14))
  Downloading packaging-24.2-py3-none-any.whl.metadata (3.2 kB)
Collecting pyparsing>=2.3.1 (from matplotlib<=3.8.4->-r ./OpenCSP/requirements.txt (line 14))
  Downloading pyparsing-3.2.1-py3-none-any.whl.metadata (5.0 kB)
Collecting python-dateutil>=2.7 (from matplotlib<=3.8.4->-r ./OpenCSP/requirements.txt (line 14))
  Downloading python_dateutil-2.9.0.post0-py2.py3-none-any.whl.metadata (8.4 kB)
Collecting tzdata>=2022.7 (from pandas>=2.2.1->-r ./OpenCSP/requirements.txt (line 15))
  Downloading tzdata-2024.2-py2.py3-none-any.whl.metadata (1.4 kB)
Collecting certifi (from pyproj>=3.6.1->-r ./OpenCSP/requirements.txt (line 17))
  Downloading certifi-2024.12.14-py3-none-any.whl.metadata (2.3 kB)
Collecting mpmath<1.4,>=1.1.0 (from sympy>=1.12->-r ./OpenCSP/requirements.txt (line 20))
  Downloading mpmath-1.3.0-py3-none-any.whl.metadata (8.6 kB)
Collecting comm>=0.1.1 (from ipykernel>=6.29.3->-r ./OpenCSP/requirements.txt (line 23))
  Downloading comm-0.2.2-py3-none-any.whl.metadata (3.7 kB)
Collecting debugpy>=1.6.5 (from ipykernel>=6.29.3->-r ./OpenCSP/requirements.txt (line 23))
  Downloading debugpy-1.8.11-cp310-cp310-win_amd64.whl.metadata (1.1 kB)
Collecting ipython>=7.23.1 (from ipykernel>=6.29.3->-r ./OpenCSP/requirements.txt (line 23))
  Downloading ipython-8.31.0-py3-none-any.whl.metadata (4.9 kB)
Collecting jupyter-client>=6.1.12 (from ipykernel>=6.29.3->-r ./OpenCSP/requirements.txt (line 23))
  Downloading jupyter_client-8.6.3-py3-none-any.whl.metadata (8.3 kB)
Collecting jupyter-core!=5.0.*,>=4.12 (from ipykernel>=6.29.3->-r ./OpenCSP/requirements.txt (line 23))
  Downloading jupyter_core-5.7.2-py3-none-any.whl.metadata (3.4 kB)
Collecting matplotlib-inline>=0.1 (from ipykernel>=6.29.3->-r ./OpenCSP/requirements.txt (line 23))
  Downloading matplotlib_inline-0.1.7-py3-none-any.whl.metadata (3.9 kB)
Collecting nest-asyncio (from ipykernel>=6.29.3->-r ./OpenCSP/requirements.txt (line 23))
  Downloading nest_asyncio-1.6.0-py3-none-any.whl.metadata (2.8 kB)
Collecting psutil (from ipykernel>=6.29.3->-r ./OpenCSP/requirements.txt (line 23))
  Downloading psutil-6.1.1-cp37-abi3-win_amd64.whl.metadata (23 kB)
Collecting pyzmq>=24 (from ipykernel>=6.29.3->-r ./OpenCSP/requirements.txt (line 23))
  Downloading pyzmq-26.2.0-cp310-cp310-win_amd64.whl.metadata (6.2 kB)
Collecting tornado>=6.1 (from ipykernel>=6.29.3->-r ./OpenCSP/requirements.txt (line 23))
  Downloading tornado-6.4.2-cp38-abi3-win_amd64.whl.metadata (2.6 kB)
Collecting traitlets>=5.4.0 (from ipykernel>=6.29.3->-r ./OpenCSP/requirements.txt (line 23))
  Downloading traitlets-5.14.3-py3-none-any.whl.metadata (10 kB)
Collecting colorama (from pytest>=8.1.1->-r ./OpenCSP/requirements.txt (line 26))
  Downloading colorama-0.4.6-py2.py3-none-any.whl.metadata (17 kB)
Collecting exceptiongroup>=1.0.0rc8 (from pytest>=8.1.1->-r ./OpenCSP/requirements.txt (line 26))
  Downloading exceptiongroup-1.2.2-py3-none-any.whl.metadata (6.6 kB)
Collecting iniconfig (from pytest>=8.1.1->-r ./OpenCSP/requirements.txt (line 26))
  Downloading iniconfig-2.0.0-py3-none-any.whl.metadata (2.6 kB)
Collecting pluggy<2,>=1.5 (from pytest>=8.1.1->-r ./OpenCSP/requirements.txt (line 26))
  Downloading pluggy-1.5.0-py3-none-any.whl.metadata (4.8 kB)
Collecting tomli>=1 (from pytest>=8.1.1->-r ./OpenCSP/requirements.txt (line 26))
  Downloading tomli-2.2.1-py3-none-any.whl.metadata (10 kB)
Collecting XlsxWriter>=0.5.7 (from python-pptx>=0.6.23->-r ./OpenCSP/requirements.txt (line 27))
  Downloading XlsxWriter-3.2.0-py3-none-any.whl.metadata (2.6 kB)
Collecting lxml>=3.1.0 (from python-pptx>=0.6.23->-r ./OpenCSP/requirements.txt (line 27))
  Downloading lxml-5.3.0-cp310-cp310-win_amd64.whl.metadata (3.9 kB)
Collecting typing-extensions>=4.9.0 (from python-pptx>=0.6.23->-r ./OpenCSP/requirements.txt (line 27))
  Downloading typing_extensions-4.12.2-py3-none-any.whl.metadata (3.0 kB)
Collecting et-xmlfile (from openpyxl>=3.1.2->-r ./OpenCSP/requirements.txt (line 29))
  Downloading et_xmlfile-2.0.0-py3-none-any.whl.metadata (2.7 kB)
Collecting pyvirtualdisplay>=1.3 (from pytest-xvfb>=3.0.0->-r ./OpenCSP/requirements.txt (line 37))
  Downloading PyVirtualDisplay-3.0-py3-none-any.whl.metadata (943 bytes)
Collecting coverage>=7.5 (from coverage[toml]>=7.5->pytest-cov>=4.1.0->-r ./OpenCSP/requirements.txt (line 38))
  Downloading coverage-7.6.10-cp310-cp310-win_amd64.whl.metadata (8.4 kB)
Collecting decorator (from ipython>=7.23.1->ipykernel>=6.29.3->-r ./OpenCSP/requirements.txt (line 23))
  Downloading decorator-5.1.1-py3-none-any.whl.metadata (4.0 kB)
Collecting jedi>=0.16 (from ipython>=7.23.1->ipykernel>=6.29.3->-r ./OpenCSP/requirements.txt (line 23))
  Downloading jedi-0.19.2-py2.py3-none-any.whl.metadata (22 kB)
Collecting prompt_toolkit<3.1.0,>=3.0.41 (from ipython>=7.23.1->ipykernel>=6.29.3->-r ./OpenCSP/requirements.txt (line 23))
  Downloading prompt_toolkit-3.0.48-py3-none-any.whl.metadata (6.4 kB)
Collecting pygments>=2.4.0 (from ipython>=7.23.1->ipykernel>=6.29.3->-r ./OpenCSP/requirements.txt (line 23))
  Downloading pygments-2.19.1-py3-none-any.whl.metadata (2.5 kB)
Collecting stack_data (from ipython>=7.23.1->ipykernel>=6.29.3->-r ./OpenCSP/requirements.txt (line 23))
  Downloading stack_data-0.6.3-py3-none-any.whl.metadata (18 kB)
Collecting platformdirs>=2.5 (from jupyter-core!=5.0.*,>=4.12->ipykernel>=6.29.3->-r ./OpenCSP/requirements.txt (line 23))
  Downloading platformdirs-4.3.6-py3-none-any.whl.metadata (11 kB)
Collecting pywin32>=300 (from jupyter-core!=5.0.*,>=4.12->ipykernel>=6.29.3->-r ./OpenCSP/requirements.txt (line 23))
  Downloading pywin32-308-cp310-cp310-win_amd64.whl.metadata (8.3 kB)
Collecting six>=1.5 (from python-dateutil>=2.7->matplotlib<=3.8.4->-r ./OpenCSP/requirements.txt (line 14))
  Downloading six-1.17.0-py2.py3-none-any.whl.metadata (1.7 kB)
Collecting parso<0.9.0,>=0.8.4 (from jedi>=0.16->ipython>=7.23.1->ipykernel>=6.29.3->-r ./OpenCSP/requirements.txt (line 23))
  Downloading parso-0.8.4-py2.py3-none-any.whl.metadata (7.7 kB)
Collecting wcwidth (from prompt_toolkit<3.1.0,>=3.0.41->ipython>=7.23.1->ipykernel>=6.29.3->-r ./OpenCSP/requirements.txt (line 23))
  Downloading wcwidth-0.2.13-py2.py3-none-any.whl.metadata (14 kB)
Collecting executing>=1.2.0 (from stack_data->ipython>=7.23.1->ipykernel>=6.29.3->-r ./OpenCSP/requirements.txt (line 23))
  Downloading executing-2.1.0-py2.py3-none-any.whl.metadata (8.9 kB)
Collecting asttokens>=2.1.0 (from stack_data->ipython>=7.23.1->ipykernel>=6.29.3->-r ./OpenCSP/requirements.txt (line 23))
  Downloading asttokens-3.0.0-py3-none-any.whl.metadata (4.7 kB)
Collecting pure-eval (from stack_data->ipython>=7.23.1->ipykernel>=6.29.3->-r ./OpenCSP/requirements.txt (line 23))
  Downloading pure_eval-0.2.3-py3-none-any.whl.metadata (6.3 kB)
Downloading h5py-3.12.1-cp310-cp310-win_amd64.whl (3.0 MB)
   ---------------------------------------- 3.0/3.0 MB 58.1 MB/s eta 0:00:00
Downloading imageio-2.36.1-py3-none-any.whl (315 kB)
Downloading matplotlib-3.8.4-cp310-cp310-win_amd64.whl (7.7 MB)
   ---------------------------------------- 7.7/7.7 MB 52.4 MB/s eta 0:00:00
Downloading pandas-2.2.3-cp310-cp310-win_amd64.whl (11.6 MB)
   ---------------------------------------- 11.6/11.6 MB 55.6 MB/s eta 0:00:00
Downloading pillow-11.1.0-cp310-cp310-win_amd64.whl (2.6 MB)
   ---------------------------------------- 2.6/2.6 MB 37.5 MB/s eta 0:00:00
Downloading pyproj-3.7.0-cp310-cp310-win_amd64.whl (6.2 MB)
   ---------------------------------------- 6.2/6.2 MB 76.5 MB/s eta 0:00:00
Downloading pytz-2024.2-py2.py3-none-any.whl (508 kB)
Downloading scipy-1.15.0-cp310-cp310-win_amd64.whl (43.9 MB)
   ---------------------------------------- 43.9/43.9 MB 68.1 MB/s eta 0:00:00
Downloading sympy-1.13.3-py3-none-any.whl (6.2 MB)
   ---------------------------------------- 6.2/6.2 MB 128.9 MB/s eta 0:00:00
Downloading ipykernel-6.29.5-py3-none-any.whl (117 kB)
Downloading opencv_contrib_python-4.10.0.82-cp37-abi3-win_amd64.whl (45.5 MB)
   ---------------------------------------- 45.5/45.5 MB 69.0 MB/s eta 0:00:00
Downloading numpy-1.26.4-cp310-cp310-win_amd64.whl (15.8 MB)
   ---------------------------------------- 15.8/15.8 MB 58.4 MB/s eta 0:00:00
Downloading pytest-8.3.4-py3-none-any.whl (343 kB)
Downloading python_pptx-1.0.2-py3-none-any.whl (472 kB)
Downloading rawpy-0.24.0-cp310-cp310-win_amd64.whl (855 kB)
   ---------------------------------------- 855.3/855.3 kB 9.6 MB/s eta 0:00:00
Downloading openpyxl-3.1.5-py2.py3-none-any.whl (250 kB)
Downloading pypylon-4.1.0-cp39-abi3-win_amd64.whl (89.5 MB)
   ---------------------------------------- 89.5/89.5 MB 67.2 MB/s eta 0:00:00
Downloading pysolar-0.13-py3-none-any.whl (46 kB)
Downloading tqdm-4.67.1-py3-none-any.whl (78 kB)
Downloading pytest_xvfb-3.0.0-py3-none-any.whl (5.6 kB)
Downloading pytest_cov-6.0.0-py3-none-any.whl (22 kB)
Downloading comm-0.2.2-py3-none-any.whl (7.2 kB)
Downloading contourpy-1.3.1-cp310-cp310-win_amd64.whl (218 kB)
Downloading coverage-7.6.10-cp310-cp310-win_amd64.whl (211 kB)
Downloading cycler-0.12.1-py3-none-any.whl (8.3 kB)
Downloading debugpy-1.8.11-cp310-cp310-win_amd64.whl (5.2 MB)
   ---------------------------------------- 5.2/5.2 MB 52.6 MB/s eta 0:00:00
Downloading exceptiongroup-1.2.2-py3-none-any.whl (16 kB)
Downloading fonttools-4.55.3-cp310-cp310-win_amd64.whl (2.2 MB)
   ---------------------------------------- 2.2/2.2 MB 20.8 MB/s eta 0:00:00
Downloading ipython-8.31.0-py3-none-any.whl (821 kB)
   --------------------------------------- 821.6/821.6 kB 11.9 MB/s eta 0:00:00
Downloading jupyter_client-8.6.3-py3-none-any.whl (106 kB)
Downloading jupyter_core-5.7.2-py3-none-any.whl (28 kB)
Downloading kiwisolver-1.4.8-cp310-cp310-win_amd64.whl (71 kB)
Downloading lxml-5.3.0-cp310-cp310-win_amd64.whl (3.8 MB)
   ---------------------------------------- 3.8/3.8 MB 11.9 MB/s eta 0:00:00
Downloading matplotlib_inline-0.1.7-py3-none-any.whl (9.9 kB)
Downloading mpmath-1.3.0-py3-none-any.whl (536 kB)
   ---------------------------------------- 536.2/536.2 kB 3.5 MB/s eta 0:00:00
Downloading packaging-24.2-py3-none-any.whl (65 kB)
Downloading pluggy-1.5.0-py3-none-any.whl (20 kB)
Downloading pyparsing-3.2.1-py3-none-any.whl (107 kB)
Downloading python_dateutil-2.9.0.post0-py2.py3-none-any.whl (229 kB)
Downloading PyVirtualDisplay-3.0-py3-none-any.whl (15 kB)
Downloading pyzmq-26.2.0-cp310-cp310-win_amd64.whl (641 kB)
   ---------------------------------------- 641.6/641.6 kB 3.4 MB/s eta 0:00:00
Downloading tomli-2.2.1-py3-none-any.whl (14 kB)
Downloading tornado-6.4.2-cp38-abi3-win_amd64.whl (438 kB)
Downloading traitlets-5.14.3-py3-none-any.whl (85 kB)
Downloading typing_extensions-4.12.2-py3-none-any.whl (37 kB)
Downloading tzdata-2024.2-py2.py3-none-any.whl (346 kB)
Downloading XlsxWriter-3.2.0-py3-none-any.whl (159 kB)
Downloading certifi-2024.12.14-py3-none-any.whl (164 kB)
Downloading colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Downloading et_xmlfile-2.0.0-py3-none-any.whl (18 kB)
Downloading iniconfig-2.0.0-py3-none-any.whl (5.9 kB)
Downloading nest_asyncio-1.6.0-py3-none-any.whl (5.2 kB)
Downloading psutil-6.1.1-cp37-abi3-win_amd64.whl (254 kB)
Downloading jedi-0.19.2-py2.py3-none-any.whl (1.6 MB)
   ---------------------------------------- 1.6/1.6 MB 27.9 MB/s eta 0:00:00
Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
Downloading prompt_toolkit-3.0.48-py3-none-any.whl (386 kB)
Downloading pygments-2.19.1-py3-none-any.whl (1.2 MB)
   ---------------------------------------- 1.2/1.2 MB 31.1 MB/s eta 0:00:00
Downloading pywin32-308-cp310-cp310-win_amd64.whl (6.6 MB)
   ---------------------------------------- 6.6/6.6 MB 80.7 MB/s eta 0:00:00
Downloading six-1.17.0-py2.py3-none-any.whl (11 kB)
Downloading decorator-5.1.1-py3-none-any.whl (9.1 kB)
Downloading stack_data-0.6.3-py3-none-any.whl (24 kB)
Downloading asttokens-3.0.0-py3-none-any.whl (26 kB)
Downloading executing-2.1.0-py2.py3-none-any.whl (25 kB)
Downloading parso-0.8.4-py2.py3-none-any.whl (103 kB)
Downloading pure_eval-0.2.3-py3-none-any.whl (11 kB)
Downloading wcwidth-0.2.13-py2.py3-none-any.whl (34 kB)
Installing collected packages: wcwidth, pywin32, pyvirtualdisplay, pytz, pypylon, pure-eval, mpmath, XlsxWriter, tzdata, typing-extensions, traitlets, tornado, tomli, sympy, six, pyzmq, pyparsing, pygments, psutil, prompt_toolkit, pluggy, platformdirs, pillow, parso, packaging, numpy, nest-asyncio, lxml, kiwisolver, iniconfig, fonttools, executing, exceptiongroup, et-xmlfile, decorator, debugpy, cycler, coverage, colorama, certifi, asttokens, tqdm, stack_data, scipy, rawpy, python-pptx, python-dateutil, pytest, pysolar, pyproj, openpyxl, opencv-contrib-python, matplotlib-inline, jupyter-core, jedi, imageio, h5py, contourpy, comm, pytest-xvfb, pytest-cov, pandas, matplotlib, jupyter-client, ipython, ipykernel
Successfully installed XlsxWriter-3.2.0 asttokens-3.0.0 certifi-2024.12.14 colorama-0.4.6 comm-0.2.2 contourpy-1.3.1 coverage-7.6.10 cycler-0.12.1 debugpy-1.8.11 decorator-5.1.1 et-xmlfile-2.0.0 exceptiongroup-1.2.2 executing-2.1.0 fonttools-4.55.3 h5py-3.12.1 imageio-2.36.1 iniconfig-2.0.0 ipykernel-6.29.5 ipython-8.31.0 jedi-0.19.2 jupyter-client-8.6.3 jupyter-core-5.7.2 kiwisolver-1.4.8 lxml-5.3.0 matplotlib-3.8.4 matplotlib-inline-0.1.7 mpmath-1.3.0 nest-asyncio-1.6.0 numpy-1.26.4 opencv-contrib-python-4.10.0.82 openpyxl-3.1.5 packaging-24.2 pandas-2.2.3 parso-0.8.4 pillow-11.1.0 platformdirs-4.3.6 pluggy-1.5.0 prompt_toolkit-3.0.48 psutil-6.1.1 pure-eval-0.2.3 pygments-2.19.1 pyparsing-3.2.1 pyproj-3.7.0 pypylon-4.1.0 pysolar-0.13 pytest-8.3.4 pytest-cov-6.0.0 pytest-xvfb-3.0.0 python-dateutil-2.9.0.post0 python-pptx-1.0.2 pytz-2024.2 pyvirtualdisplay-3.0 pywin32-308 pyzmq-26.2.0 rawpy-0.24.0 scipy-1.15.0 six-1.17.0 stack_data-0.6.3 sympy-1.13.3 tomli-2.2.1 tornado-6.4.2 tqdm-4.67.1 traitlets-5.14.3 typing-extensions-4.12.2 tzdata-2024.2 wcwidth-0.2.13
Chocolatey v2.4.1
Installing the following packages:
ffmpeg
By installing, you accept licenses for the packages.
Downloading package from source 'https://community.chocolatey.org/api/v2/'

Progress: Downloading ffmpeg 7.1.0... %
<snip>
```


### python script that produces the error

```python
#!/bin/bash
cd OpenCSP/opencsp
set PYTHONPATH="$PWD\.."
echo "$PYTHONPATH"
$env:PYTHONPATH="$PWD\.."
pytest --color=yes -rs -vv --cov=. --cov-report term --cov-config=.coveragerc
```


### Relevant console output

```shell
================================== FAILURES ===================================
______________ TestCalibrateDisplayShape.test_xy_screen_fraction ______________

self = <opencsp.app.sofast.test.test_CalibrateDisplayShape.TestCalibrateDisplayShape testMethod=test_xy_screen_fraction>

    @pytest.mark.skipif(os.name != 'nt', reason='Does not pass in Linux environment for unkonwn reason.')
    def test_xy_screen_fraction(self):
        """Tests xy points"""
        data_meas = self.cal.get_data()
>       np.testing.assert_allclose(
            data_meas['xy_screen_fraction'].data, self.data_exp['pts_xy_screen_fraction'], rtol=0, atol=1e-6
        )

app\sofast\test\test_CalibrateDisplayShape.py:81: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (<function assert_allclose.<locals>.compare at 0x0000018431D0F7F0>, array([[0.01      , 0.01989899, 0.02979798, ..., 0...2,
        0.99      ],
       [0.99      , 0.99      , 0.99      , ..., 0.01      , 0.01      ,
        0.01      ]]))
kwds = {'equal_nan': True, 'err_msg': '', 'header': 'Not equal to tolerance rtol=0, atol=1e-06', 'verbose': True}

    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError: 
E           Not equal to tolerance rtol=0, atol=1e-06
E           
E           (shapes (2, 8806), (2, 8808) mismatch)
E            x: array([[0.01    , 0.019899, 0.029798, ..., 0.960303, 0.970202, 0.99    ],
E                  [0.99    , 0.99    , 0.99    , ..., 0.01    , 0.01    , 0.01    ]])
E            y: array([[0.01    , 0.019899, 0.029798, ..., 0.960303, 0.970202, 0.99    ],
E                  [0.99    , 0.99    , 0.99    , ..., 0.01    , 0.01    , 0.01    ]])

C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\contextlib.py:79: AssertionError
______________ TestCalibrateDisplayShape.test_xyz_screen_coords _______________

self = <opencsp.app.sofast.test.test_CalibrateDisplayShape.TestCalibrateDisplayShape testMethod=test_xyz_screen_coords>

    @pytest.mark.skipif(os.name != 'nt', reason='Does not pass in Linux environment for unkonwn reason.')
    def test_xyz_screen_coords(self):
        """Tests xyz points"""
        data_meas = self.cal.get_data()
>       np.testing.assert_allclose(
            data_meas['xyz_screen_coords'].data, self.data_exp['pts_xyz_screen_coords'], rtol=0, atol=1e-6
        )

app\sofast\test\test_CalibrateDisplayShape.py:89: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (<function assert_allclose.<locals>.compare at 0x0000018431D0C160>, array([[-1.70363486e+00, -1.66911226e+00, -1.63478... [ 3.22877882e-03,  3.14730465e-03,  3.09913363e-03, ...,
         1.22931971e-03,  1.03224241e-03,  1.13443250e-03]]))
kwds = {'equal_nan': True, 'err_msg': '', 'header': 'Not equal to tolerance rtol=0, atol=1e-06', 'verbose': True}

    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError: 
E           Not equal to tolerance rtol=0, atol=1e-06
E           
E           (shapes (3, 8806), (3, 8808) mismatch)
E            x: array([[-1.703635e+00, -1.669112e+00, -1.634790e+00, ...,  1.605272e+00,
E                    1.639982e+00,  1.708847e+00],
E                  [ 1.704457e+00,  1.704338e+00,  1.704222e+00, ..., -1.711925e+00,...
E            y: array([[-1.703635e+00, -1.669112e+00, -1.634790e+00, ...,  1.605272e+00,
E                    1.639982e+00,  1.708847e+00],
E                  [ 1.704457e+00,  1.704338e+00,  1.704222e+00, ..., -1.711925e+00,...

C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\contextlib.py:79: AssertionError
```


### Contribution Guidelines

- [X] I agree to follow this project's contribution guidelines

--- a/requirements.txt
+++ b/requirements.txt
@@ -16,7 +16,7 @@ pandas >= 2.2.1
 pillow >= 10.2.0
 pyproj >= 3.6.1
 pytz >= 2024.1
-scipy >= 1.12.0
+scipy >= 1.12.0, <= 1.14.1
 sympy >= 1.12
 
 # packages that can't be installed with conda but can be installed with pip on solo #pippkgs

